src/components/Header/TopMainNav/SearchBar/SearchBar.tsx

```diff
@@ -1,10 +1,32 @@
-import React, { useRef, useState } from 'react';
+import React, { ChangeEvent, useRef, useState } from 'react';
+import { useLocation } from '@reach/router';
+import { useSearch } from 'src/hooks';
 import { SearchIcon } from '.';
 
 export const SearchBar = () => {
-  const [searchTerm, setSearchTerm] = useState('');
+  const location = useLocation();
+  const initialSearch = new URLSearchParams(location.search).get('q');
+
+  const [searchTerm, setSearchTerm] = useState(initialSearch?.toString());
```

Comment:
I don't think you need this, `useSearch` will return `state` which has `query`?

Reply:
We have a controlled component, so we need that for our input field. I'll have a look if that'll work. 🤔 👍

Reply:
Our `useSearch` hook doesn't return a `setSearch` function that would work with a `query` the same way as `useState` hook would, since it's a controlled component, I don't think we can go without `useState`.

I can do it a different way with a `useRef` tho. So we'll grab an input value with `inputRef.current.value` when `onKeyDown` and trigger `search()` function then, if you prefer. 🙂

Reply:
hey @snikidev , I must admit, I don't follow :) perhaps we can get on a call?

To summerise what I mean - state is internally held by the hook, so `setSearch` -> `search` and `searchTerm` is `query`

Reply:
There should be no need for `useRef` or `useState` here, the hook abstracts away the actions and state of a "search", just without providing a component.